### DISCUSSION TITLE
Can't expand blogs while fetching a single user

### DISCUSSION THREAD

**Discussion by albeniskerqeli10** [pocketbase/pocketbase#2915] [Q&A] [answered]
Hi there, I'm having trouble expanding user blogs when fetching a single user.

Here's my code;
Single User Request:
 ```
useEffect(() => {
    const getUser = async () => {
      const user = await pb.collection('users').getOne(id as string, {
        expand: 'blogs',
      });
      if (user) {
        setUser(user);
      }
    };
    getUser();
  }, [id]);
```
users collection at admin ui:
![Capture](https://github.com/pocketbase/pocketbase/assets/58092596/3e080867-3d48-425c-a065-588ba1aaa62f)

blogs collection:
![Capture](https://github.com/pocketbase/pocketbase/assets/58092596/d5b6df8b-2c77-4e3b-9dd7-b40e015ec960)

Response is like this:
![Capture](https://github.com/pocketbase/pocketbase/assets/58092596/dbbe744d-e84c-495d-8600-515ae8a637ab)

Expanding the user object when fetching a single blog (or multiple) works fine, but it doesn't work the same way when trying to expand all blogs that belong to a single user.

**Comment by ganigeorgiev** [comment]
Make sure that the client request satisfy the View API rule of the `blogs` collection.

**Reply by albeniskerqeli10** [reply]
I didn't set any rule for the View API, so it's empty and everyone has access to it.

**Reply by ganigeorgiev** [reply]
For the `blogs` collection? Keep in mind that by default the rules are "locked" aka. only admins can access.

Could you paste your collections configuration as JSON (you can export it from the Admin UI > Settings > Export collections)?

**Reply by ganigeorgiev** [reply]
Oh, actually now I see that in your user response screenshot you don't have any `blogs` relations - notice the `blogs` array is empty.

Do you have any `blogs` items associated in the user record?

**Reply by albeniskerqeli10** [reply]
```
[
    {
        "id": "mb1lyf7k17yk4xz",
        "name": "blogs",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "wb7lqakh",
                "name": "title",
                "type": "text",
                "system": false,
                "required": true,
                "options": {
                    "min": null,
                    "max": null,
                    "pattern": ""
                }
            },
            {
                "id": "zzl5kiwe",
                "name": "content",
                "type": "text",
                "system": false,
                "required": true,
                "options": {
                    "min": null,
                    "max": null,
                    "pattern": ""
                }
            },
            {
                "id": "0az06oey",
                "name": "image",
                "type": "text",
                "system": false,
                "required": false,
                "options": {
                    "min": null,
                    "max": null,
                    "pattern": ""
                }
            },
            {
                "id": "2mwzsko1",
                "name": "user",
                "type": "relation",
                "system": false,
                "required": true,
                "options": {
                    "collectionId": "_pb_users_auth_",
                    "cascadeDelete": false,
                    "minSelect": null,
                    "maxSelect": 1,
                    "displayFields": [
                        "id",
                        "username",
                        "avatar",
                        "created"
                    ]
                }
            },
            {
                "id": "nw7sks7q",
                "name": "likes",
                "type": "json",
                "system": false,
                "required": false,
                "options": {}
            }
        ],
        "indexes": [],
        "listRule": "",
        "viewRule": "",
        "createRule": "user = @request.auth.id",
        "updateRule": "",
        "deleteRule": "user = @request.auth.id",
        "options": {}
    },
    {
        "id": "_pb_users_auth_",
        "name": "users",
        "type": "auth",
        "system": false,
        "schema": [
            {
                "id": "eetljxa0",
                "name": "avatar",
                "type": "url",
                "system": false,
                "required": false,
                "options": {
                    "exceptDomains": null,
                    "onlyDomains": null
                }
            },
            {
                "id": "mrv7nvjm",
                "name": "blogs",
                "type": "relation",
                "system": false,
                "required": false,
                "options": {
                    "collectionId": "mb1lyf7k17yk4xz",
                    "cascadeDelete": false,
                    "minSelect": null,
                    "maxSelect": null,
                    "displayFields": []
                }
            }
        ],
        "indexes": [],
        "listRule": "",
        "viewRule": "@request.auth.id != \"\"",
        "createRule": "",
        "updateRule": "id = @request.auth.id",
        "deleteRule": "id = @request.auth.id",
        "options": {
            "allowEmailAuth": true,
            "allowOAuth2Auth": true,
            "allowUsernameAuth": false,
            "exceptEmailDomains": null,
            "manageRule": null,
            "minPasswordLength": 8,
            "onlyEmailDomains": null,
            "requireEmail": false
        }
    }
]
```

**Reply by ganigeorgiev** [reply]
@albeniskerqeli10 The rules are not the issue, please check my previous comment. You don't seem to have any `blogs` records associated to the user.

The relation fields from the 2 collections are not "connected/synced" with one-another.
Even if you have associated a user in the `blogs.user` field, it will not appear magically in the `users.blogs` field. The relation fields have their own separate value(s).

**Reply by ganigeorgiev** [reply, marked as answer]
If that's the case, my recommendation would be to keep the relation between the `users` and `blogs` only in a single place (eg. in `blogs` or in a separate junction collection).

For example, if the relation between the 2 collections is kept only in the `blogs` collection, then you can use [indirect expand](https://pocketbase.io/docs/expanding-relations/#indirect-expand) from the `users` collection to retrieve all blogs that belongs to the specific user:

```js
await pb.collection("users").getOne(id, {
    expand: "blogs(user)"
})
```

**Reply by albeniskerqeli10** [reply]
Yeah, it's basically the same case. Each blog is linked to a user, but when I attempt to retrieve blogs from the user collection it doesn't work and it appears empty. I considered updating `user.blogs` at the same time when creating a blog, but the method you demonstrated with indirect expand works perfectly. Thank you a lot!
